flow/scripts/generate_abstract.tcl

```diff
@@ -1,16 +1,21 @@
 source $::env(SCRIPTS_DIR)/load.tcl
-load_design 6_final.odb 6_final.sdc "Starting generation of abstract views"
 
-read_spef $::env(RESULTS_DIR)/6_final.spef
-set_propagated_clock [all_clocks]
+set stem [expr {[info exists ::env(ABSTRACT_SOURCE)] ? $::env(ABSTRACT_SOURCE) : "6_final"}]
+
+load_design $stem.odb $stem.sdc "Starting generation of abstract views"
+
+if {[file exists $::env(RESULTS_DIR)/$stem.spef]} {
+  read_spef $::env(RESULTS_DIR)/$stem.spef
+  set_propagated_clock [all_clocks]
+}
```

Comment:
you should call estimate_parasitics otherwise

Reply:
Can you give a bit more detail?

Like?

```
else {
   estimate_parasitics
}
```